lj/lj.go

```diff
@@ -18,17 +18,28 @@
 // Package lj implements common lumberjack types and functions.
 package lj
 
+import (
+	"crypto/tls"
+)
+
 // Batch is an ACK-able batch of events as has been received by lumberjack
 // server implemenentations. Batches must be ACKed, for the server
 // implementations returning an ACK to it's clients.
 type Batch struct {
-	Events []interface{}
-	ack    chan struct{}
+	ack        chan struct{}
+	TLS        *tls.ConnectionState // TLS connection metadata. Nil for non-TLS connections.
+	SourceAddr string               // Source address of the connection.
+	Events     []interface{}
 }
 
 // NewBatch creates a new ACK-able batch.
-func NewBatch(evts []interface{}) *Batch {
-	return &Batch{evts, make(chan struct{})}
+func NewBatch(evts []interface{}, remoteAddr string, tlsState *tls.ConnectionState) *Batch {
```

Comment:
Doc addition for `remoteAddr` and `tlsState`?

Also, this is a breaking change. Maybe add as `NewBatchTLS` and have `NewBatch(evts []... { return NewBatchTLS(evts, "", nil) }`.

Reply:
Changed to `NewBatchWithSourceMetadata`.